week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -1,25 +1,80 @@
 class MusicPlayer {
-    // Add your code here
-
+  // Add your code here
+  constructor() {
+    this.playlist = {
+      songs: [],
+      currentSong: 0,
+    };
+  }
+
+  add(song, artist) {
+    this.playlist.songs.push([`${song} by ${artist}`]);
+  }
+  play() {
+    if (this.playlist.songs.length === 0) {
+      console.log("Playlist is empty! Please add a song!");
+    } else {
+      console.log(
+        `Currently playing: ${this.playlist.songs[this.playlist.currentSong]}`
+      );
+    }
+  }
+  skip() {
+    if (this.playlist.songs.length - 1 <= this.playlist.currentSong) {
+      this.playlist.currentSong = 0;
+      console.log(
+        `Currently playing: ${this.playlist.songs[this.playlist.currentSong]}`
+      );
+    } else {
+      this.playlist.currentSong++;
+      console.log(
+        `Currently playing: ${this.playlist.songs[this.playlist.currentSong]}`
+      );
+    }
+  }
+  previous() {
+    this.playlist.currentSong--;
+    console.log(
+      `Currently playing: ${this.playlist.songs[this.playlist.currentSong]}`
+    );
```

Comment:
What happens here if you are on the first song and you try to go to the previous song?

You could add more error handling here.